src/public/js/rs.js

```diff
@@ -1,16 +1,33 @@
 'use strict';
 var RS = (function() {
-  // (actually ROT-13 right now, as a dummy example. We can replace w/ RS)
-  var ROT = 13;
+  // Using a RS with 256 symbol size 255 symobls in a codeword and 245 message symbols.
+  // n-k = 10 code symbols. Can correct up to 5 symbol errors in each 255 symbol block
+  var n = 255;
+  var k = 245;
 
   function encode(msg) {
-    // dummy cipher for now -- advances each letter in message by ROT
-    return UTIL.shift(msg, ROT);
+    // Does Reed Solomon decoding on k length message block of symbols.
+    // Cast msg to Uint8Array
+    // If msg is not length k then pad with 0s.
+    var rsMsg = new Uint8Array(k);
+    for (var i = 0; i < msg.length; i++) {
+      rsMsg[i] = msg[i].charCodeAt();
+    }
+    return encodeRSBlock(rsMsg, n, k);
   }
 
   function decode(msg) {
-    // dummy cipher for now -- advances each letter in message by -ROT
-    return UTIL.shift(msg, -ROT);
+    // Decode and correct an n length RS codeword in to a k length message.
+    // Cast input codeword to Uint8Array
+    var decoded = decodeRSBlock(msg, n, k);
+    var decodedString = new Array(k);
+    for (var i = 0; i < k; i++) {
+      if(decoded[i] == 0) {
+        break;
```

Comment:
A 0 is just the end of the message essentially. So I break the loop because there is no more need to continue. This was somewhat of a design decision that we should feel free to discuss! I made the conclusion that our message will never contain char = 0 unless it is padding at the end of the last block. Since most strings never contain the null character I think this is a decent decision, but we could decide otherwise and do something like trim trailing 0s on the final output. Or just print all the 0 chars (not sure how the printing handles that).